R/init.R

```diff
@@ -222,27 +186,246 @@ init <- function(data,
+  }
+  res
+}
+
+#' Add a Footer to a `teal` App
+#'
+#' @description Adds a footer to the `teal` app.
+#'
+#' @param app (`environment`) The `teal` app environment.
+#' @param footer (`shiny.tag` or `character(1)`) The footer content to set. Defaults to an empty paragraph tag.
+#' @export
+#' @examples
+#' app <- init(
+#'   data = teal_data(IRIS = iris),
+#'   modules = modules(example_module())
+#' ) |>
+#'   modify_footer("Custom footer")
+#'
+#' shinyApp(app$ui, app$server)
+modify_footer <- function(app, footer = tags$p()) {
+  res <- app
+  res$ui <- function(request, ...) {
+    args <- list(...)
+    args$footer <- footer
+    ui_tq <- do.call(app$ui, c(list(request = request), args)) |>
+      htmltools::tagQuery()
+    ui_tq$find("#teal-footer")$replaceWith(tags$div(id = "teal-footer", args$footer))$allTags()
+  }
+  res
+}
+
+#' Add a Landing Popup to a `teal` App
+#'
+#' @description Adds a landing popup to the `teal` app. This popup will be shown when the app starts.
+#'
+#' This modifier is used to display a popup dialog when the application starts.
+#' The dialog blocks access to the application and must be closed with a button before the application can be viewed.
+#'
+#' @param app (`environment`) The `teal` app environment.
+#' @param id (`character(1)`) The ID of the modal dialog.
+#' @param label (`character(1)`) Label of the module.
+#' @param title (`character(1)`) Text to be displayed as popup title.
+#' @param content (`character(1)`, `shiny.tag` or `shiny.tag.list`) with the content of the popup.
+#'  Passed to `...` of `shiny::modalDialog`. See examples.
+#' @param buttons (`shiny.tag` or `shiny.tag.list`) Typically a `modalButton` or `actionButton`. See examples.
+#' @export
+#' @examples
+#' app <- init(
+#'   data = teal_data(IRIS = iris, MTCARS = mtcars),
+#'   modules = modules(example_module())
+#' ) |>
+#'   add_landing_popup(
+#'     title = "Welcome",
+#'     content = "This is a landing popup.",
+#'     buttons = modalButton("Accept")
+#'   )
+#'
+#' shinyApp(app$ui, app$server)
+add_landing_popup <- function(
+    app,
+    id = "landingpopup",
+    title = NULL,
+    content = NULL,
+    buttons = modalButton("Accept")) {
+  old_server <- app$server
+
+  app$server <- function(input, output, session) {
+    old_server(input, output, session)
+    showModal(
+      modalDialog(
+        id = id,
+        title = title,
+        content,
+        footer = buttons
+      )
+    )
+  }
+  app
+}
+
+#' Add a Custom Server Logic to a `teal` App
+#'
+#' @description Adds a custom server function to the `teal` app. This function can define additional server logic.
+#'
+#' @param app (`environment`) The `teal` app environment.
+#' @param custom_server (`function`) The custom server function to set.
+#' @export
+#' @examples
+#' app <- init(
+#'   data = teal_data(IRIS = iris),
+#'   modules = modules(example_module())
+#' ) |>
```

Comment:
pipe available from R 4.1

Reply:
I remember we had a discussion about this earlier. Did we decide on bumping the R version? It's already over 3 years since `|>` is available.

Reply:
If you'd like to bump up R in the Description please link [related issue](https://github.com/insightsengineering/teal/issues/1143) and put issue on the board. I would accept this change thanks to [this comment](https://github.com/insightsengineering/teal/issues/1268#issuecomment-2406719999)